internal/dataplane/parser/translators/ingress.go

```diff
@@ -387,3 +393,57 @@ func MaybePrependRegexPrefixForIngressV1Fn(ingress *netv1.Ingress, applyLegacyHe
 		return lo.ToPtr(MaybePrependRegexPrefix(path, regexPrefix, applyLegacyHeuristic))
 	}
 }
+
+func generateRewriteURIConfig(path string) (string, error) {
+	var out []rune
+	in := []rune(path)
+	for i := 0; i < len(in); i++ {
```

Comment:
I think manipulating strings by moving the index front and back is hard to understand and error prone. How about finding out places of all `$` s and processing them part by part? Like 
```golang
    // use strings.Split to split the path by $
    parts := strings.Split(path,"$")
    configParts := []string{}
    for i, part := range parts {
        if i == 0 {
            // initial part. This part is not after a $, so we do not run the replacing.
             configParts = append(configParts, part)
             continue
        }

            // i > 0 means this part is after a '$'. 
           // replace the beginning digits to uriCapture patterns. (like 1ab -> (uri_captures[1])ab )
            replacedPart, err := replaceDigitsAfterURICaptureMark(part)
            if err != nil {
                return "", err
            }
           // add replaced Part to configParts
          configParts = append(configParts, replacedPart)
    }
    // re-join the replaced segements by '$'.
    return strings.Join(configParts,"$")
```
Or like this if we want to keep the places of $:
```golang
parts := []string{}
index := 0
for index < len(path) {
    i := strings.IndexRune(path[index:],'$')
    if i != -1 {
        // If we want to keep the position, we may define a structure to store the indices of '$'.
        parts = append(parts, path[index:i])
        index = i
    } else {
     // i == -1 means that there are no $ in the remaining part, we should end the loop here.
        break
    }
}
// process `parts` ...
```

Reply:
It just occurred to me whether we should deal with escaping ？

Reply:
yes